SmartDeviceLink/private/SDLLifecycleManager.m

```diff
@@ -192,9 +192,29 @@ - (instancetype)initWithConfiguration:(SDLConfiguration *)configuration delegate
 
     _backgroundTaskManager = [[SDLBackgroundTaskManager alloc] initWithBackgroundTaskName:BackgroundTaskTransportName];
 
+    if (SDL_SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"14")) {
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(sdl_accessoryDisconnected:)
+                                                     name:EAAccessoryDidDisconnectNotification
+                                                   object:nil];
+    }
+    
     return self;
 }
 
+- (void)sdl_accessoryDisconnected:(NSNotification *)notification {
+    if (SDL_SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"14")) {
+        EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
+        SDLLogD(@"Accessory with serial number: %@, and connectionID: %lu disconnecting.", accessory.serialNumber, (unsigned long)accessory.connectionID);
+        if (self.backgroundTaskManager != nil) {
+            [self.backgroundTaskManager endBackgroundTask];
+        }
+        if (self.secondaryTransportManager != nil) {
+            [_secondaryTransportManager endBackgroundTask];
+        }
+    }
+}
+
```

Comment:
I have not looked into the code if ackground tasks are ended elsewhere. I have the impression the iOS bug might be caused by task expiration not being handled properly. So the idea to end them when not needed anymore is good. However ending them without being in sync with the disconnect process is dangerous as the thread might survive and run at a 100% whenever the app's process comes back to life.